### DISCUSSION TITLE
How to use lodash to filter an array of proxy objects in alpine js.

### DISCUSSION THREAD

**Discussion by ppshobi** [alpinejs/alpine#689] [4. General]
`main.js`

```
window._ = require('lodash');
import 'alpinejs'
```
and in my template I try to use lodash to get the selected `plan` from the `region()` method

```
<script>
        window.data = function data() {
          return {
            plans: [], // array of plans
            plan: null, // this is the selected plans slug
            region() {
              let plan = _.findIndex(this.plans, {slug: this.plan}) // here I try to use the lodash to find the currently selected plan based on the slug property
              console.log(plan): // -1 , even though the object with such slug exists
            },
          }
        }
</script>
```
But I believe because the `plans` array is not a plain array rather is an array of `Proxy` objects, So the lodash is failing to apply the filter on it. How can I filter objects from an array of proxy objects?

**Comment by SimoTod** [comment]
Proxy act as the underlying object most of the time so it should work. See https://codepen.io/SimoTod/pen/yLOByKa
Maybe check that plans contains the expected array.

As a separate note, in 2020 you probably don't need loadash anymore because arrays natively support most of those functions (you only need polyfill for IE11 if you support it), For example: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex
